README.md

```diff
@@ -51,66 +51,84 @@ npm install ramldt2jsonschema --save
+  } catch (err) {
+    console.log(err)
+    return
+  }
+  console.log('#%RAML 1.0 Library\n')
+  console.log(yaml.safeDump(raml, { 'noRefs': true }))
+}
+
+main()
 ```
 
-#### js2dt
+#### Resolving references
 
+When the input contains external references (`!include`, `uses:`, `$ref`, etc.) and the referred files are not in the same directory as the script it is being ran from, you may provide a third argument `basePath` to both `dt2js` and `js2dt`. All references will then be resolved relative to that base path.
+
+Example of using `basePath` argument in dt2js:
 ```js
+// Script below ran from /home/john/where/ever/
+// Reference is located at /home/john/schemas/simple_person.json
 const raml2json = require('ramldt2jsonschema')
-const join = require('path').join
-const fs = require('fs')
 
-const filePath = join(__dirname, 'schema.json')
-const jsonData = fs.readFileSync(filePath).toString()
+const ramlStr = `
+  #%RAML 1.0 Library
 
-const raml = raml2json.js2dt(jsonData, 'Person')
-
-console.log(`#%RAML 1.0 Library\n${yaml.safeDump({ types: raml }, {'noRefs': true})}`)
+  types:
+    Person: !include simple_person.json
+`
+const basePath = '/home/john/schemas/' // or '../../schemas/'
+const schema = raml2json.dt2js(ramlStr, 'Person', basePath)
+console.log(JSON.stringify(schema, null, 2))
 ```
 
 ### Limitations
 
-- in js2dt,
-  - the following JSON schema properties are not supported and as a result, will not be converted:
-    - `not` (or `false` when used as a schema boolean)
-    - object's `dependencies`
-    - array's `additionalItems`
-    - the `propertyNames` keyword
-    - the `contains` keyword
-  - JSON schema number and integer's `exclusiveMaximum` and `exclusiveMinimum` properties are converted to raml 'maximum' and 'minimum'
-  - the JSON schema `oneOf` property is processed the same way as the JSON schema `anyOf` property, it is converted to the RAML `union` type
-  - the JSON schema `type: 'string'` with `media` and `binaryEncoding: 'binary'` is converted to the RAML `file` type
-  - the JSON schema `type: string` with a `pattern` that matches exactly one of the RAML date types will be converted in the matching RAML date type
-  - the JSON schema `media` property with `anyOf`, elements of which have the property `mediaType` will be converted to `fileTypes` in RAML
-
-- in dt2js,
-  - the following RAML properties are not supported/converted:
-    - annotations
+- in js2dt
+  - the following JSON Schema properties are not supported and as a result, may not be converted as expected:
+    > dependencies, exclusiveMaximum, exclusiveMinimum, items (array value), allOf, oneOf, not, format (email, hostname, ipv4, ipv6, uri), readOnly
+  - the following JSON Schema properties won't be converted at all:
+    > $schema, additionalItems, contains, id, $id, propertyNames, definitions, links, fragmentResolution, media, pathStart, targetSchema
+- in dt2js
+  - types with `type: nil` aren't supported and won't be converted
```

Comment:
why not?

Reply:
Few things to consider:
1. Both Libraries and regular API docs should be supported;
2. Types must be resolved before conversion to not output types with refs to in-existing types;
3. When document is resolved, all types are moved from the root, to places they are used at, thus types which are not used are deleted (AMF logic).

So when I was adding libraries support in a recent commits, I had to patch libraries docs by adding fake endpoint and using converted type in it so it doesn't disappear after resolution. While doing that it turned out that custom types with `type: nil` which are used by an endpoint are resolved to a nameless inline declaration `type: nil`, thus it's impossible to find them by name.

I'll try to add more magic to handle this specific case.

Reply:
I've reported disappearing `nil` types issue on AMF github: https://github.com/aml-org/amf/issues/555. Let's first see if this is an expected behaviour.

Reply:
let's track this in an issue on this repo so we can release this new version of ramldt2jsonschema and address this issue later

Reply:
Ok. Here's an issue https://github.com/raml-org/ramldt2jsonschema/issues/104